### ISSUE TITLE
LanguageTool is not working in the comment textarea

### ISSUE THREAD

**Issue by Defman21** [forem/forem#13859]
**Describe the bug**

LanguageTool (and probably Grammarly) is not working when I'm writing a comment. The textarea has the `spellcheck` attribute, and it's set to `false` (why?).

**To Reproduce**

1. Install LanguageTool.
2. Go to any post.
3. Try to write a comment with some grammar mistakes.

**Expected behavior**

LanguageTool catches them all.

**Desktop (please complete the following information):**

- OS, version: macOS 10.15
- Browser, version: Firefox 88

**Comment by Link2Twenty** [comment]
Grammarly was disabled due to this bug #13814, I believe the aim is to find the cause and reenable Grammarly when possible.

**Comment by cmgorton** [comment]
Hey @Defman21 are you having trouble using LanguageTool on DEV or a different Forem? Our team tested LanguageTool on DEV in Firefox, Edge, and Chrome. It works on all three browsers for our team.

**Comment by cmgorton** [comment]
Ok. We were testing this in new posts not comments. LanguageTool does not seem to trigger in the comments like mentioned in this issue. We are investigating more now.

**Comment by Defman21** [comment]
Yeah, posts work, comments doesn't :(

**Comment by aitchiss** [comment]
I had a quick investigation and I can confirm that this is happening as a result of substituting the original textarea for the MentionAutocomplete textarea(s). I tried even specifying directly on the textarea `spellcheck="true"` but for some reason it's being overridden by the LanguageTool extension and set to `spellcheck="false"`

LanguageTool [is open source](https://github.com/languagetool-org/languagetool) so we might need to dig into their codebase to see if there is a way to prevent this happening.

**Comment by Defman21** [comment]
AFAIK, the web extension is not open source.

**Comment by aitchiss** [comment]
> AFAIK, the web extension is not open source.

ah sorry you're right @Defman21 - I was looking at an outdated extension repo 🙁

**Comment by bernd-wechner** [comment]
> I had a quick investigation and I can confirm that this is happening as a result of substituting the original textarea for the MentionAutocomplete textarea(s). I tried even specifying directly on the textarea `spellcheck="true"` but for some reason it's being overridden by the LanguageTool extension and set to `spellcheck="false"`
> 
> LanguageTool [is open source](https://github.com/languagetool-org/languagetool) so we might need to dig into their codebase to see if there is a way to prevent this happening.

That's odd. I landed here on the tail end of an issue I filed with Language Tool (thinking I'd have to report this, but here it is already reported) and here is their reply:

> They have set a spellcheck=false on the text field which disallows any spell checker to run there.

So I checked. Went to dev.to and on the front page clicked Comment, and a text box opens, and yep, Language Tool not there in that textbox.

So I inspect the element in the browser and I see:

```
<textarea data-gramm_editor="false" data-mention-autocomplete-active="true" placeholder="Add to the discussion" onfocus="handleFocus(event)" onkeyup="handleKeyUp(event)" onkeydown="handleKeyDown(event)" oninput="handleChange(event)" id="text-area" required="required" class="crayons-textfield comment-textarea crayons-textfield--ghost" aria-label="Add a comment to the discussion" name="comment[body_markdown]" spellcheck="false" data-gramm="false" style="transition: none 0s ease 0s; height: 128px;"></textarea>
```

Note: it contains in that mess of attributes `spellcheck="false"`.

I remove that in the Inspector (using Firefox dev tools here, Chrome works same though) so it now reads:

```
<textarea data-gramm_editor="false" data-mention-autocomplete-active="true" placeholder="Add to the discussion" onfocus="handleFocus(event)" onkeyup="handleKeyUp(event)" onkeydown="handleKeyDown(event)" oninput="handleChange(event)" id="text-area" required="required" class="crayons-textfield comment-textarea crayons-textfield--ghost" aria-label="Add a comment to the discussion" name="comment[body_markdown]" data-gramm="false" style="transition: none 0s ease 0s; height: 128px;"></textarea>
```

If you look closely, the only change is I removed `spellcheck="false"`.

Now I got to the comment box and Language Tools are there, available and working. 

Seems a simple issue to me, nothing to explore, forem simply needs to drop `spellcheck="false"` from the comment boxes.

For completeness, I repeated this exercise but did not remove it, instead replaced it with `spellcheck="true"`. But get the same result, Language Tools do start working. So I'm not sure what @aitchiss did differently or got wrong there, because that one attribute most certainly does what it claims to do and what Language Tools elects to honour. In fact, they write:

> We could ignore it. I'm not sure though if this would break anything

And so it seems this is a trivial thing for forem to fix. It would be awesome if you did as Language Tools rocks, it is free and just plain awesome and will improve the quality of comments across the site if people adopt it. It has improved the quality of my comments all over the web for a while now. Simply prompting me to fix my typos before I post and doing it in such an elegant way.

**Comment by aitchiss** [comment]
Sorry if my original comment wasn't clear - what I meant was that we're _not_ explicitly setting spellcheck to "false" in our code (I've tried searching the whole repo for any mention of spellcheck 😅 

I've also observed mixed results with when it appears - sometimes `spellcheck="false"` gets added to the comment area, and sometimes it doesn't - e.g. maybe 1 in 4 times when I enter a comment area I see the HTML:

```html
<textarea data-gramm_editor="false" data-mention-autocomplete-active="true" placeholder="Add to the discussion" 
  onfocus="handleFocus(event)" onkeyup="handleKeyUp(event)" onkeydown="handleKeyDown(event)" 
  oninput="handleChange(event)" id="text-area" required="required" class="crayons-textfield comment-textarea crayons- 
  textfield--ghost" aria-label="Add a comment to the discussion" name="comment[body_markdown]" style="transition: 
  none 0s ease 0s; height: 128px;"></textarea>
```

In my previous exploration, I incorrectly attributed the difference to the LanguageTool extension, but I can see now that often this `spellcheck="false"` pops up even when the tool is disabled/removed.

TL;DR this requires some more investigation to figure out when and how this attribute is being added.

**Comment by bernd-wechner** [comment]
Thanks for clarifying. That does sound mysterious. If this code base makes no mention, then almost perforce it implicates a dependency. But I won't jump the gun ad try to diagnose something  you're already trying to, without at least some investigative work ;-).

**Comment by aitchiss** [comment]
> If this code base makes no mention, then almost perforce it implicates a dependency

For sure! The Reach UI combobox was my first suspicion (we use it to power the mention autocomplete in the text area), but I searched their code (very briefly) for `spellcheck` too and couldn't see it 🙃  Definitely time for some proper detective work... 

Thanks for drawing our attention to this again though 🌟

**Comment by Defman21** [comment]
![image](https://user-images.githubusercontent.com/7100645/153398742-77022493-1436-41c0-ab6d-0f03f6f87eea.png)


Looks like LanguageTool is the one who sets `spellcheck=false` for the textarea. Using Firefox, I've set to break on attribute changes, then clicked on the textarea to focus it.

**Comment by Defman21** [comment]
Though, without the extension there's still this attribute called `data-gramm_editor` which is set to `false`. LT might observe this attribute as a fallback to check if it should disable itself for some sites that use it to disable Grammarly ([because some people want it to be disabled](https://stackoverflow.com/questions/37444906/how-to-stop-extensions-add-ons-like-grammarly-on-contenteditable-editors))

**Comment by bernd-wechner** [comment]
Where is that screenshot from? The source file? If it's identifiable and belongs to LanguageTool, letting them know here:
https://github.com/languagetool-org/languagetool/issues/6314
will help.

**Comment by bernd-wechner** [comment]
Oh, and far as I can tell LT is open source:

https://github.com/languagetool-org/languagetool

Grammarly I'm not so sure of.

**Comment by nickytonline** [comment]
`data-gramm_editor` is a custom attribute we added that disables Grammarly because the Grammarly extension updated at some point last year and stopped working. The Grammarly team was looking into this. See #13814.

**Comment by Defman21** [comment]
@bernd-wechner the web extension is closed-source as far as I know.

**Comment by Defman21** [comment]
So here's the response from the LT issue linked above:

>It appears that our extension is excluding itself from the text field by setting spellcheck="false". This happens because forem is cloning the textarea that initially received the user's focus. The initial textarea is then removed from the dom. This confuses our add-on. We will make the add-on more robust to beter handle such cases.

Should I close this issue or let it stay open just so other people could find it?
